### DISCUSSION TITLE
WebGPU Library

### DISCUSSION THREAD

**Discussion by ygdrasil-io** [littlektframework/littlekt#264] [General]
Hello,

I've been working on WebGPU bindings for a while now, with the ultimate goal of creating a Korge backend using it. You can check out the project here: https://github.com/wgpu4k/wgpu4k

I'm reaching out to see if you'd be interested in collaborating on a shared library dedicated to WebGPU that could be used across various projects. I recently spoke with a contributor from https://github.com/openrndr who also seemed interested.

The idea is to combine our efforts to build a reliable and robust binding. By working together, we can leverage each other's experience and avoid duplicating efforts.

**Comment by LeHaine** [comment]
Hey!

I do recall seeing your message in Slack and I did check it out before. A pure WebGPU bindings module & library would be great! It does make sense to combine efforts to prevent duplicate work. I am all for it.

Currently, I have all the WebGPU bindings baked into the core module inside a single package for LittleKt. It already seems your bindings have almost everything needed in LittleKt which is great. Would love to see getting WebGPU bindings for mobile, somehow, in the future 😃.

**Reply by ygdrasil-io** [reply]
Nice,

Right now, I'm prioritizing native targets, including iOS. Android is a bit trickier because it requires JNI.

I found this app that uses Rust to expose methods to Android and utilizes WebGPU under the hood. It's not directly usable, but I think it could be a good starting point: https://github.com/ygdrasil-io/ruffle-android

If you have some spare time, I'd appreciate any feedback on what functionalities or features would be missing from wgpu4k for using it with Littlekt in its current state.

**Reply by LeHaine** [reply]
Awesome. I'd need to get some time to go through it all and see but initially I saw that the Adapter & Device limits were missing (which shouldn't be a huge add or anything crazy) but that was just at first glance. 

And agreed on the Android. I was tinkering with attempting to generate the JNI bindings using what LJWGL3 uses but it's a lot more complex than I expected 😄. I didn't have time to build my own generator for it yet but I do have an open branch for Android support for WGPU.

**Comment by ygdrasil-io** [comment]
I hate JNI, that's now a fact, but Android will be supported on the next release :

![image](https://github.com/user-attachments/assets/9646f3b2-ce3a-4002-af51-828813cf445a)

**Reply by LeHaine** [reply]
Amazing. Great work! I'll need to start integrating the library, hopefully soon. Exciting!
